packages/app/hooks/useQueue.ts

```diff
@@ -0,0 +1,32 @@
+import { API } from "@template/api-client";
+import { QueuePartial } from "@template/common";
+import useSWR, { responseInterface } from "swr";
+
+const TEN_SECONDS_IN_MS = 100000;
+const FIFTEEN_SECOND_IN_MS = 150000;
+
+type queueResponse = responseInterface<QueuePartial, any>;
+
+interface UseQueueReturn {
+  queue: queueResponse["data"];
+  queuesError: queueResponse["error"];
+  mutateQueue: queueResponse["mutate"];
+}
+
+export function useQueue(qid: number): UseQueueReturn {
+  const { data: queue, error: queuesError, mutate: mutateQueue } = useSWR(
+    qid && `/api/v1/queues/${qid}`,
+    async () => API.queues.get(Number(qid)),
+    {
+      refreshInterval: Math.floor(
```

Comment:
Could someone explain why the `refreshInterval` is needed here but not before?

Reply:
The queue is what we want to be polling constantly since it's dynamic. Questions aren't really updated by anyone but the student, so we probably don't need to refresh them.

Reply:
Gotcha. 
I thought it had something to do with it being put in the hook but that clears it up